gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -425,6 +434,47 @@ public static List<Service> parseService(
         Transport.GRPC);
   }
 
+  private static Optional<List<String>> getInclusionMethodListFromServiceYaml(
+      Optional<com.google.api.Service> serviceYamlProtoOpt, String protoPackage) {
+    if (!serviceYamlProtoOpt.isPresent()
+        || serviceYamlProtoOpt.get().getPublishing().getLibrarySettingsCount() == 0) {
+      return Optional.empty();
+    }
+    List<ClientLibrarySettings> librarySettingsList =
+        serviceYamlProtoOpt.get().getPublishing().getLibrarySettingsList();
+    // verify library_settings.version matches with proto package name
+    // This should be validated in upstream. Give warnings if not match.
+    if (!librarySettingsList.get(0).getVersion().isEmpty()
```

Comment:
This is nice but I think we should just ignore this case. If `v1.foo.bar` is configured and the proto package is `v2`, we will continue generate `foo.bar` because the full name does not match.

Reply:
summary from off-line discussion: will keep this check for logging purpose. This logic won't change behavior of output (yes, will include all when not matching).